database/models/materials-model.js

```diff
@@ -0,0 +1,24 @@
+const db = require(`../dbConfig`);
+
+const find = (materialId = undefined) => {
+  return db(`materials`).modify(qb => {
+    if (materialId) {
+      qb.where({ material_id: materialId });
+    }
+  });
+};
```

Comment:
So you are encapsulating data you found by using the material_id filter you made in getAllMaterials, but what is the materialId assigned undefined for?

Reply:
That sets it to undefined if you don't supply a value for that parameter. Doing it this way lets us have a single function for find that returns all entries by default but can also take an ID to only return a specific record.

Reply:
That's awesome, thanks for explaining. We can probably use a variable and if statement on the getAllMaterials in a new query so frontend can request by id then. That will keep our codebase much leaner.